### ISSUE TITLE
Couldn't insert the nullable data with the MySQL database engine.

### ISSUE THREAD

**Issue by BayoNet** [ClickHouse/ClickHouse#5697]
**Describe the bug**
ClickHouse throws the exception 

```
Received exception from server (version 19.10.1):
Code: 1000. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: mysqlxx::BadQuery: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'float, float_nullable) VALUES (2,3,4,5)' at line 1 (localhost:3306). 
```

when inserting data into MySQL with the MySQL database engine. #5599.

**How to reproduce**

I used:

- MySQL server 5.7
- ClickHouse 19.10.1 revision 54422

I created a table in MySQL

```
mysql> CREATE TABLE `test`.`test` (
    ->   `int_id` INT NOT NULL AUTO_INCREMENT,
    ->   `int_nullable` INT NULL DEFAULT NULL,
    ->   `float` FLOAT NOT NULL,
    ->   `float_nullable` FLOAT NULL DEFAULT NULL,
    ->   PRIMARY KEY (`int_id`));
Query OK, 0 rows affected (0,09 sec)

mysql> insert into test (`int_id`, `float`) VALUES (1,2);
Query OK, 1 row affected (0,00 sec)

mysql> select * from test;
+--------+--------------+-------+----------------+
| int_id | int_nullable | float | float_nullable |
+--------+--------------+-------+----------------+
|      1 |         NULL |     2 |           NULL |
+--------+--------------+-------+----------------+
1 row in set (0,00 sec)
```

Then I started `clickhouse-client` and tried to execute the following queries:

```
CREATE DATABASE mysql_db ENGINE = MySQL('localhost:3306', 'test', 'bayonet', '123')
```
```
SHOW DATABASES

┌─name─────┐
│ default  │
│ mysql_db │
│ system   │
└──────────┘
```
```
SHOW TABLES FROM mysql_db

┌─name─┐
│ test │
└──────┘
```
```
SELECT *
FROM mysql_db.test 

┌─int_id─┬─int_nullable─┬─float─┬─float_nullable─┐
│      1 │         ᴺᵁᴸᴸ │     2 │           ᴺᵁᴸᴸ │
└────────┴──────────────┴───────┴────────────────┘
```
```
INSERT INTO mysql_db.test values (2,1,3,4)

INSERT INTO mysql_db.test VALUES

Received exception from server (version 19.10.1):
Code: 1000. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: mysqlxx::BadQuery: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'float, float_nullable) VALUES (1,2,3,4)' at line 1 (localhost:3306). 

1 rows in set. Elapsed: 0.002 sec. 
```

Then I created another table in MySQL without of nullable fields and ClickHouse didn't throw exception at insertion of data into this new nullable-free table.

**Comment by zhang2014** [comment]
fixed in #5704 . BTW, this is because MySQL table contains a column and it's named float

**Comment by blinkov** [comment]
cc @zhang2014
